src/EnergyPlus/MixedAir.cc

```diff
@@ -230,6 +230,127 @@ namespace MixedAir {
 
 	// Functions
 
+	Real64 OAGetFlowRate(int OAPtr)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Simon Vidanovic
+		//       DATE WRITTEN   March 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE
+		// Return flow rate [m3/s] for current controller
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		Real64 FlowRate(0);
+
+		if ((OAPtr > 0) && (OAPtr <= NumOAControllers) && (StdRhoAir != 0))
+		{
+			FlowRate = OAController(OAPtr).OAMassFlow / StdRhoAir;
+		}
+
+		return FlowRate;
+	};
+
+	Real64 OAGetMinFlowRate(int OAPtr)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Simon Vidanovic
+		//       DATE WRITTEN   March 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE
+		// Return min flow rate [m3/s] for current controller
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		Real64 MinFlowRate(0);
+
+		if ((OAPtr > 0) && (OAPtr <= NumOAControllers))
+		{
+			MinFlowRate = OAController(OAPtr).MinOA;
+		}
+
+		return MinFlowRate;
+	};
+
+	void OASetDemandManagerVentilationState(int OAPtr, bool aState)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Simon Vidanovic
+		//       DATE WRITTEN   March 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE
+		// Setting if controller is in demand manager state
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		if ((OAPtr > 0) && (OAPtr <= NumOAControllers))
+		{
+			OAController(OAPtr).ManageDemand = aState;
+		}
+	};
+
+	void OASetDemandManagerVentilationFlow(int OAPtr, Real64 aFlow)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Simon Vidanovic
+		//       DATE WRITTEN   March 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE
+		// Set demand manager flow rate for use in managing demand use
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		if ((OAPtr > 0) && (OAPtr <= NumOAControllers))
+		{
+			OAController(OAPtr).DemandLimitFlowRate = aFlow * StdRhoAir;
+		}
+
+	};
+
+	int GetOAController(std::string const & OAName)
```

Comment:
I think we actually do. Take a look at other routines that are passing std::string. They all use "&" so that string is passed as reference and not copy created. I think that idea behind this was not to make copies of strings since that is expensive operation.
Also, seems that every routine in EnergyPlus is transferred from fortran in this way. Please confirm this. Thanks!